nemo/collections/common/data/lhotse/nemo_adapters.py

```diff
@@ -332,14 +332,16 @@ def __iter__(self) -> Generator[Cut, None, None]:
 
         # Handle NeMo tarred manifests with offsets.
         # They have multiple JSONL entries where audio paths end with '-sub1', '-sub2', etc. for each offset.
-        offset_pattern = re.compile(r'^.+(-sub\d+)$')
+        offset_pattern = re.compile(r'^(?P<stem>.+)(?P<sub>-sub\d+)(?P<ext>\.\w+)?$')
```

Comment:
at some documentation here. what does this re do? what's <stem> here

Reply:
`(?P<group_name>[actual pattern])` syntax creates named groups. You can refer to groups via their names rather than indexes, like 

```python
m = pattern.match(string)
m.group("group_name")  # instead of m.group(1) or 2 or 3
```